File: contracts/responses/PlaylistResponses.ts
import { Playlist } from "../models/Playlist";
import { PlaylistItem } from "../models/PlaylistItem";
import { PageInfo } from "./Pagination";

export interface PlaylistResponse {
  playlist: Playlist;
  items: PlaylistItem[];
}

export interface PlaylistListResponse {
  items: Playlist[];
  pageInfo: PageInfo;
}

File: contracts/responses/PlaylistResponses.kt
package contracts.responses

import contracts.models.Playlist
import contracts.models.PlaylistItem
import kotlinx.serialization.Serializable

@Serializable
data class PlaylistResponse(
    val playlist: Playlist,
    val items: List<PlaylistItem> = emptyList()
)

@Serializable
data class PlaylistListResponse(
    val items: List<Playlist> = emptyList(),
    val pageInfo: PageInfo
)
